be/src/olap/tablet.h

```diff
@@ -242,6 +242,8 @@ class Tablet : public std::enable_shared_from_this<Tablet> {
 
     void build_tablet_report_info(TTabletInfo* tablet_info);
 
+    OLAPStatus clone_tablet_meta(TabletMetaSharedPtr new_tablet_meta);
```

Comment:
What locks does caller need to hold? one or two?
And why not implement as a TabletMeta static function? which like
static copy(const TabletMeta& src, TabletMeta* dest)?

If the function name is clone, I don't know the direction of clone operation.

Reply:
And you should comment for this function why we need it other than get it from RocksDB

Reply:
Need tablet header lock.
Could not implement as a static method because need tablet header lock.

Reply:
In your implementation, there is no lock.

Reply:
Callers acquire the header lock.

Reply:
In our implementation, many methods in tablet not have header lock but the callers get tablet header lock and then call the method.